frappe/core/doctype/user/user.py

```diff
@@ -789,7 +789,7 @@ def get_all_roles():
 @frappe.whitelist()
 def get_roles(arg=None):
 	"""get roles for a user"""
-	return frappe.get_roles(frappe.form_dict["uid"])
+	return frappe.get_roles(frappe.form_dict.get("uid", frappe.session.user))
```

Comment:
Alternatively, we could just remove the uid param, and always let get_roles take the session user. But this would break anyone else who passes a uid other than the current sessions user. 🤔
```suggestion
	return frappe.get_roles()
```